apis/s3/v1alpha2/bucketpolicy_types.go

```diff
@@ -48,147 +47,133 @@ type BucketPolicyParameters struct {
 	// BucketNameSelector selects a reference to an S3Bucket to retrieve its bucketName
 	// +optional
 	BucketNameSelector *runtimev1alpha1.Selector `json:"bucketNameSelector,omitempty"`
-
-	// UserName presents the name of the IAM user this BucketPolicy is concerned with.
-	// +optional
-	UserName *string `json:"userName,omitempty"`
-
-	// UserNameRef references to an S3Bucket to retrieve its userName
-	// +optional
-	UserNameRef *runtimev1alpha1.Reference `json:"userNameRef,omitempty"`
-
-	// UserNameSelector selects a reference to an S3Bucket to retrieve its userName
-	// +optional
-	UserNameSelector *runtimev1alpha1.Selector `json:"userNameSelector,omitempty"`
-}
-
-// Serialize is the custom marshaller for the BucketPolicyParameters
-func (p *BucketPolicyParameters) Serialize() (interface{}, error) {
-	m := make(map[string]interface{})
-	m["Version"] = p.PolicyVersion
-	if p.PolicyID != "" {
-		m["Id"] = p.PolicyID
-	}
-	slc := make([]interface{}, len(p.PolicyStatement))
-	for i, v := range p.PolicyStatement {
-		msg, err := v.Serialize()
-		if err != nil {
-			return nil, err
-		}
-		slc[i] = msg
-	}
-	m["Statement"] = slc
-	return m, nil
 }
 
 // BucketPolicyStatement defines an individual statement within the
 // BucketPolicyBody
 type BucketPolicyStatement struct {
 	// Optional identifier for this statement, must be unique within the
 	// policy if provided.
-	StatementID string `json:"sid,omitempty"`
+	// +optional
+	StatementID *string `json:"sid,omitempty"`
 
 	// The effect is required and specifies whether the statement results
 	// in an allow or an explicit deny. Valid values for Effect are Allow and Deny.
+	// +kubebuilder:validation:Enum=Allow;Deny
 	Effect string `json:"effect"`
 
 	// Used with the S3 policy to specify the principal that is allowed
 	// or denied access to a resource.
+	// +optional
 	Principal *BucketPrincipal `json:"principal,omitempty"`
 
 	// Used with the S3 policy to specify the users which are not included
 	// in this policy
+	// +optional
 	NotPrincipal *BucketPrincipal `json:"notPrincipal,omitempty"`
 
 	// Each element of the PolicyAction array describes the specific
 	// action or actions that will be allowed or denied with this PolicyStatement.
+	// +optional
 	PolicyAction []string `json:"action,omitempty"`
```

Comment:
We should match the Go field names with corresponding jsontag unless there is a big reason for not doing so.